week05/clause_test.go

```diff
@@ -0,0 +1,60 @@
+package demo
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestClausesString(t *testing.T) {
+	t.Parallel()
+
+	tcases := []struct {
+		name    string
+		clauses Clauses
+		want    string
+	}{
+		{"empty", Clauses{}, ""},
```

Comment:
Always use the name of the field when initializing it. If the order of the fields change, or a new field is added, your code will fail to compile. You can avoid this by using the field name in the initialization.

Also, the zero value of `string` is `""`, you don't need to put that in your test cases.

```suggestion
		{name: "empty", clauses: Clauses{}},
```